### ISSUE TITLE
GPU training

### ISSUE THREAD

**Issue by syn66** [NVIDIA/retinanet-examples#253]
When I use my own data set for training, the GPU occupancy rate is very low. How to solve this problem

**Comment by yashnv** [comment]
What GPU are you using? Which branch and docker container are you using to train your model?

**Comment by syn66** [comment]
I use 2080ti , branch 20.03 and pytorch 20.08 docker container to train my model, GPU-utility Vary between 1% and 20%.
![1603073021653](https://user-images.githubusercontent.com/32169725/96393846-f11e8880-11f2-11eb-8ee4-441124d05228.jpg)

**Comment by syn66** [comment]
![IMG_20201019_095848 HEIC](https://user-images.githubusercontent.com/32169725/96393893-18755580-11f3-11eb-907b-dce2c641c44e.jpg)

**Comment by syn66** [comment]
I use 2080ti , branch 20.03 and pytorch 20.08 docker container to train my model, GPU-utility Vary between 1% and 20%.
![IMG_20201019_101258](https://user-images.githubusercontent.com/32169725/96394127-d26cc180-11f3-11eb-99a4-d65da432d460.jpg)

**Comment by syn66** [comment]
> What GPU are you using? Which branch and docker container are you using to train your model? Also, what is your GPU utility, can you post the output of `nvidia-smi` during training?
I use 2080ti , branch 20.03 and pytorch 20.08 docker container to train my model, GPU-utility Vary between 1% and 20%.
![IMG_20201019_101258](https://user-images.githubusercontent.com/32169725/96408542-4fa92e00-1216-11eb-811b-358b3c19272a.jpg)

**Comment by syn66** [comment]
When I train the model, the following interface is not displayed.

[  566/30000] focal loss: 1.101, box loss: 0.369, 0.322s/10-batch (fw: 0.123s, bw: 0.195s), 31.1 im/s, lr: 0.0003
[  757/30000] focal loss: 1.075, box loss: 0.346, 0.315s/10-batch (fw: 0.115s, bw: 0.196s), 31.8 im/s, lr: 0.00039
[  944/30000] focal loss: 1.015, box loss: 0.357, 0.322s/10-batch (fw: 0.122s, bw: 0.196s), 31.0 im/s, lr: 0.00047
[ 1128/30000] focal loss: 1.119, box loss: 0.377, 0.327s/10-batch (fw: 0.127s, bw: 0.196s), 30.6 im/s, lr: 0.0005
[ 1316/30000] focal loss: 1.056, box loss: 0.353, 0.319s/10-batch (fw: 0.117s, bw: 0.195s), 31.3 im/s, lr: 0.0005
[ 1505/30000] focal loss: 1.087, box loss: 0.346, 0.319s/10-batch (fw: 0.119s, bw: 0.196s), 31.4 im/s, lr: 0.0005
[ 1690/30000] focal loss: 1.106, box loss: 0.370, 0.326s/10-batch (fw: 0.126s, bw: 0.196s), 30.7 im/s, lr: 0.0005
[ 1883/30000] focal loss: 1.042, box loss: 0.339, 0.312s/10-batch (fw: 0.112s, bw: 0.196s), 32.0 im/s, lr: 0.0005
[ 2074/30000] focal loss: 1.036, box loss: 0.347, 0.316s/10-batch (fw: 0.115s, bw: 0.196s), 31.7 im/s, lr: 0.0005
[ 2257/30000] focal loss: 1.126, box loss: 0.377, 0.329s/10-batch (fw: 0.129s, bw: 0.196s), 30.4 im/s, lr: 0.0005
[ 2445/30000] focal loss: 1.053, box loss: 0.345, 0.319s/10-batch (fw: 0.117s, bw: 0.195s), 31.3 im/s, lr: 0.0005
[ 2635/30000] focal loss: 1.047, box loss: 0.334, 0.317s/10-batch (fw: 0.117s, bw: 0.196s), 31.6 im/s, lr: 0.0005
[ 2819/30000] focal loss: 1.110, box loss: 0.374, 0.327s/10-batch (fw: 0.127s, bw: 0.195s), 30.6 im/s, lr: 0.0005
[ 3012/30000] focal loss: 1.020, box loss: 0.332, 0.312s/10-batch (fw: 0.112s, bw: 0.196s), 32.0 im/s, lr: 0.0005
[ 3203/30000] focal loss: 1.021, box loss: 0.342, 0.315s/10-batch (fw: 0.114s, bw: 0.196s), 31.8 im/s, lr: 0.0005
[ 3386/30000] focal loss: 1.112, box loss: 0.367, 0.328s/10-batch (fw: 0.128s, bw: 0.196s), 30.5 im/s, lr: 0.0005
[ 3574/30000] focal loss: 1.058, box loss: 0.338, 0.320s/10-batch (fw: 0.118s, bw: 0.195s), 31.2 im/s, lr: 0.0005
[ 3764/30000] focal loss: 1.081, box loss: 0.349, 0.318s/10-batch (fw: 0.117s, bw: 0.196s), 31.5 im/s, lr: 0.0005
[ 3948/30000] focal loss: 1.100, box loss: 0.360, 0.327s/10-batch (fw: 0.127s, bw: 0.195s), 30.6 im/s, lr: 0.0005
[ 4134/30000] focal loss: 1.041, box loss: 0.345, 0.324s/10-batch (fw: 0.122s, bw: 0.196s), 30.9 im/s, lr: 0.0005
[ 4298/30000] focal loss: 1.016, box loss: 0.324, 0.366s/10-batch (fw: 0.140s, bw: 0.199s), 27.3 im/s, lr: 0.0005
[ 4463/30000] focal loss: 1.097, box loss: 0.363, 0.365s/10-batch (fw: 0.159s, bw: 0.198s), 27.4 im/s, lr: 0.0005
[ 4650/30000] focal loss: 0.988, box loss: 0.295, 0.322s/10-batch (fw: 0.120s, bw: 0.195s), 31.1 im/s, lr: 0.0005
[ 4840/30000] focal loss: 1.051, box loss: 0.345, 0.317s/10-batch (fw: 0.117s, bw: 0.196s), 31.5 im/s, lr: 0.0005
[ 5025/30000] focal loss: 1.080, box loss: 0.349, 0.325s/10-batch (fw: 0.125s, bw: 0.196s), 30.8 im/s, lr: 0.0005
[ 5213/30000] focal loss: 1.065, box loss: 0.349, 0.320s/10-batch (fw: 0.120s, bw: 0.196s), 31.2 im/s, lr: 0.0005
[ 5406/30000] focal loss: 1.010, box loss: 0.328, 0.312s/10-batch (fw: 0.111s, bw: 0.196s), 32.1 im/s, lr: 0.0005
[ 5592/30000] focal loss: 1.085, box loss: 0.361, 0.324s/10-batch (fw: 0.123s, bw: 0.196s), 30.9 im/s, lr: 0.0005
[ 5775/30000] focal loss: 1.111, box loss: 0.373, 0.329s/10-batch (fw: 0.129s, bw: 0.196s), 30.4 im/s, lr: 0.0005
[ 5964/30000] focal loss: 1.051, box loss: 0.341, 0.318s/10-batch (fw: 0.116s, bw: 0.195s), 31.4 im/s, lr: 0.0005
[ 6152/30000] focal loss: 1.062, box loss: 0.343, 0.320s/10-batch (fw: 0.119s, bw: 0.196s), 31.3 im/s, lr: 0.0005
[ 6338/30000] focal loss: 1.080, box loss: 0.355, 0.324s/10-batch (fw: 0.124s, bw: 0.196s), 30.9 im/s, lr: 0.0005
[ 6530/30000] focal loss: 1.025, box loss: 0.339, 0.313s/10-batch (fw: 0.112s, bw: 0.196s), 32.0 im/s, lr: 0.0005
[ 6720/30000] focal loss: 1.033, box loss: 0.341, 0.317s/10-batch (fw: 0.116s, bw: 0.196s), 31.6 im/s, lr: 0.0005
[ 6902/30000] focal loss: 1.112, box loss: 0.373, 0.330s/10-batch (fw: 0.129s, bw: 0.196s), 30.3 im/s, lr: 0.0005
[ 7090/30000] focal loss: 1.060, box loss: 0.347, 0.321s/10-batch (fw: 0.118s, bw: 0.195s), 31.2 im/s, lr: 0.0005
[ 7280/30000] focal loss: 1.043, box loss: 0.330, 0.317s/10-batch (fw: 0.117s, bw: 0.196s), 31.6 im/s, lr: 0.0005
[ 7462/30000] focal loss: 1.106, box loss: 0.373, 0.330s/10-batch (fw: 0.130s, bw: 0.196s), 30.3 im/s, lr: 0.0005
[ 7655/30000] focal loss: 1.010, box loss: 0.332, 0.311s/10-batch (fw: 0.111s, bw: 0.196s), 32.1 im/s, lr: 0.0005
[ 7845/30000] focal loss: 1.020, box loss: 0.337, 0.316s/10-batch (fw: 0.116s, bw: 0.196s), 31.6 im/s, lr: 0.0005

**Comment by dketterer** [comment]
Maybe your CPU preprocessing is too slow? Are you using lots of small images? Try increasing the worker number of the DataLoader (in the DataIterator).

**Comment by james-nvidia** [comment]
@syn66 Have you solved this problem?

**Comment by yashnv** [comment]
Closing due to inactivity.
